src/mmap.rs

```diff
@@ -181,8 +182,20 @@ impl GuestRegionMmap {
         }
     }
 
-    fn as_volatile_slice(&self) -> VolatileSlice {
-        self.mapping.as_volatile_slice()
+    /// Convert an absolute address into an address space (GuestMemory)
+    /// to a host pointer, or return None if it is out of bounds.
+    pub fn get_host_address(&self, addr: MemoryRegionAddress) -> Option<*mut u8> {
+        // Safe because we've just range-checked addr using check_address.
+        self.check_address(addr)
+            .map(|addr| unsafe { self.as_ptr().offset(addr.raw_value() as isize) })
```

Comment:
On second thought, this looks potentially fishy for systems where `sizeof(isize) <= 32`, because we could have a legit memory region larger than `isize::MAX` bytes. Does this look like a real concern?

Otherwise, all good.

Reply:
I meant `<= 4` :(

Reply:
Yes, it should use `wrapping_offset`. It is not even unsafe, though I'm not sure why...